### PR TITLE
add support for error handling in streamSource

### PR DESCRIPTION
**Pull request by gvaradarajan**

### CODE REVIEW

File: image_source_stream.go [hunk 1]
```diff
@@ -4,8 +4,26 @@ import (
...
+	if frameErr != nil {
+		Logger.Debugw("error getting frame", "error", frameErr)
+		return true
+	}
+	return false
+}
+
 // streamSource will stream a source of images forever to the stream until the given context tells it to cancel.
-func streamSource(ctx context.Context, once func(), is ImageSource, stream Stream) {
+func streamSource(ctx context.Context, once func(), is ImageSource, stream Stream, errHandler ErrorHandler) {
```
Comment (edaniels) [inline image_source_stream.go:26]: just make this a function, not an interface.

**Review by edaniels — CHANGES_REQUESTED** [review]

File: image_source_stream.go (hunk 1 above)
Comment (gvaradarajan) [reply image_source_stream.go:26]: so I was going to do that, but the issue is that the current logic from the other PR was only introducing wait for _consecutive_ error counts. Only a struct could keep track of this, which is why I was opting for an interface. Is there some other way I should accommodate for this in the design?

Comment (edaniels) [reply image_source_stream.go:26]: a function can do that with a closure

Comment (edaniels) [reply image_source_stream.go:26]: a func returning a func

Comment (gvaradarajan) [reply image_source_stream.go:26]: yeah, fair enough. Will change

**Review by maximpertsov — APPROVED** [review]

**Pull request merged**